File: Study Plan/Top Interview 150/0028 Find the Index of the First Occurrence in a String/FindTheIndexOfTheFirstOccurrenceInAString.cs
/*  
    Problem
    --------------------------------------------------------------------------------
    28. Find the Index of the First Occurrence in a String
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/solutions/4384285/c-beats-96-in-runtime-time-complexity-o-n-space-complexity-o-1/

    - Runtime 37 ms (Beats 96.74% of users with C#)
    - Memory 37.71 MB (Beats 38.11% of users with C#)
*/
public class Solution
{
    public int StrStr(string haystack, string needle)
    {
        if (haystack.Contains(needle, StringComparison.Ordinal))
        {
            return haystack.IndexOf(needle, StringComparison.Ordinal);
        }
        else
        {
            return -1;
        }
    }
}

File: Study Plan/Top Interview 150/0028 Find the Index of the First Occurrence in a String/FindTheIndexOfTheFirstOccurrenceInAString.java
/*  
    Problem
    --------------------------------------------------------------------------------
    28. Find the Index of the First Occurrence in a String
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/solutions/4432832/java-time-complexity-o-m-n-space-complexity-o-1-runtime-0-ms-beats-100/

    - Runtime 0 ms (Beats 100.00% of users with Java)
    - Memory 41.36 MB (Beats 54.51% of users with Java)
*/

class Solution {
    public int strStr(String haystack, String needle) {
        if (haystack.contains(needle)) {
            return haystack.indexOf(needle);
        } else {
            return -1;
        }
    }
}
